depthcharge/transformers/analytes.py

```diff
@@ -0,0 +1,434 @@
+"""Transformer models for peptides and small molecules."""
+import torch
+
+from .. import utils
+from ..encoders import PositionalEncoder
+from ..mixins import ModelMixin, TransformerMixin
+from ..tokenizers import MoleculeTokenizer, PeptideTokenizer
+
+
+class _AnalyteTransformer(torch.nn.Module, ModelMixin, TransformerMixin):
+    """A transformer base class for peptide sequences.
+
+    Parameters
+    ----------
+    n_tokens : int or PeptideTokenizer
+        The number of tokens used to tokenize peptide sequences.
+    d_model : int
+        The latent dimensionality to represent the amino acids in a peptide
+        sequence.
+    positional_encoder : PositionalEncoder or bool
+        The positional encodings to use for the amino acid sequence. If
+        ``True``, the default positional encoder is used. ``False`` disables
+        positional encodings, typically only for ablation tests.
+    max_charge : int
+        The maximum precursor charge to embed.
```

Comment:
**question:** Now that we're venturing into small molecule land, do we support negative charges as well? I assume we currently also default to protonated charges only?

Reply:
**issue:** This doesn't actually seem to be a parameter in the code.

Reply:
Ah yes! This now must be handled by the user with the `global_token_hook`.